src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java

```diff
@@ -451,6 +456,10 @@ else if (request.getStartRelative() != null)
 		{
 			throw new BeanValidationException(new SimpleConstraintViolation("start_time", "relative or absolute time must be set"), "query");
 		}
+
+		if (m_metricTiersConfig.getQueryDistanceHoursLimit() > 0) {
+			startTime = Math.max(startTime, now - m_metricTiersConfig.getQueryDistanceHoursLimit() * 60 * 60_000);
```

Comment:
hmm maybe `getQueryDistanceHoursLimit` could return `datapoints_ttl` by default, and then we wouldn't need the other `Math.max` call few lines later

Reply:
I don't think this is semantically correct. `datapoints_ttl` does not relate to distance limits